pkg/server/container_create.go

```diff
@@ -201,7 +202,7 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 		specOpts = append(specOpts, containerd.WithUsername(username))
 	}
 	// Set apparmor profile, (privileged or not) if apparmor is enabled
-	if c.config.EnableAppArmor {
+	if c.config.EnableAppArmor && runcapparmor.IsEnabled() {
```

Comment:
I want to rethink this code logic a bit.  For example, we should probably be throwing an error if they request a specific localhost/profile and we ignore that request because enableapparmor is off. I feel like we are abandoning the contract then. Need more detail on the requirement :-)

Reply:
Make a lot of sense. Only when apparmor profile is not specified, we should check whether apparmor is supported and choose to or not to set default profile. Will update.

Reply:
Done.